teps/0072-results-json-serialized-records.md

```diff
@@ -0,0 +1,465 @@
+-->
+
+### User Experience (optional)
+
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be task and pipeline editors, they may trigger task and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+### Performance (optional)
+
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+- A benefit of moving to pure-JSON types is we can take advantage of JSON-aware
+  features in databases such as
+  [jsonb](https://github.com/tektoncd/results/issues/99). While this is not a
+  primary goal, it's an added benefit that would allow us to support indexed queries in the future.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+
+If it's helpful to include workflow diagrams or any other related images,
+add them under "/teps/images/". It's upto the TEP author to choose the name
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+### cel-go compatibility
+
+Currently the Results API relies on [cel-go](https://github.com/google/cel-go)
+for List filtering. Moving to an opaque byte string breaks some of the dynamic
+query behavior that we have today.
+
+To work around this, we will model the opaque byte string as a
+[google.protobuf.Struct](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#struct)
+internally for filtering. This will allow us to preserve a similar filtering
+behavior with out requiring users to adopt the `Struct` type.
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+This is primarily a backend change so our existing test infrastructure should be
+sufficient. We likely want to add additional tests to test the transition.
```

Comment:
Definitely. We could backup the data in `dogfooding` and test the migration there.